sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/executeImmediate.scala

```diff
@@ -88,6 +88,10 @@ class SubstituteExecuteImmediate(val catalogManager: CatalogManager)
           throw QueryCompilationErrors.invalidExecuteImmediateVariableType(varReference.dataType)
         }
 
+        if (varReference.eval(null) == null) {
```

Comment:
consider storing `varReference` in variable so we don't evalulate multiple times